pytorch_lightning/trainer/trainer.py

```diff
@@ -2014,13 +2004,13 @@ def optimizers(self) -> List[Optimizer]:
 
     @optimizers.setter
     def optimizers(self, new_optims: Optional[List[Optimizer]]) -> None:
-        # Necessary to rewrap optimizers to lightning
-        # They will be re-created when accessing
-        # the `lightning_optimizers` trainer property
-        self._lightning_optimizers = None
-
         self.strategy.optimizers = new_optims
 
+    @property
+    def lightning_optimizers(self) -> Dict[int, LightningOptimizer]:
+        # FIXME: why do we expose this?
```

Comment:
Only two places accesss this property:

- `OptimizerLoop._optimizer.step`
- `LightningModule.optimizers`

So I guess we could deprecate it, yay or nay?